swift/apple/FirezoneKit/Sources/FirezoneKit/Views/MenuBar.swift

```diff
@@ -451,8 +452,14 @@ public final class MenuBar: NSObject, ObservableObject {
   }
 
   private func populateOtherResourcesMenu(_ newOthers: [Resource]) {
-    otherResourcesMenuItem.isHidden = newOthers.isEmpty
-    otherResourcesSeparatorMenuItem.isHidden = newOthers.isEmpty
+    if (newOthers.isEmpty) {
+      removeItemFromMenu(menu: menu, item: otherResourcesMenuItem)
+      removeItemFromMenu(menu: menu, item: otherResourcesSeparatorMenuItem)
+    } else {
+      let i = menu.index(of: aboutMenuItem)
+      addItemToMenu(menu: menu, item: otherResourcesMenuItem, at: i)
+      addItemToMenu(menu: menu, item: otherResourcesSeparatorMenuItem, at: i + 1)
+    }
 
     // Update the menu in place so everything won't vanish if it's open when it updates
     let diff = (newOthers).difference(
```

Comment:
I wonder how the indexes stay valid when we're iterating over the difference list

Reply:
I had the same thought, maybe it's recalc'd on each iteration.